jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/PrecompressedHttpContent.java

```diff
@@ -51,20 +51,19 @@ public HttpField getETag()
     @Override
     public String getETagValue()
     {
-        //return _content.getResource().getWeakETag(_format.getEtagSuffix());
-        return null;
+        return EtagUtil.calcWeakEtag(_content.getResource().getPath(), _format.getEtagSuffix());
```

Comment:
This would only work with path-backed resources. Shouldn't we be capable of calculating the etag of a potential `MemoryResource`?

Reply:
Fixing etag support